internal/store/driver.go

```diff
@@ -642,8 +852,416 @@ func (d *Driver) calculateHTTPSEdges() map[string]ingressv1alpha1.HTTPSEdge {
+			err := d.handleHTTPHeaderFilter(filter.RequestHeaderModifier, &inboundActions, nil)
+			if err != nil {
+				return nil, err
+			}
+		case gatewayv1.HTTPRouteFilterResponseHeaderModifier:
+			err := d.handleHTTPHeaderFilter(filter.ResponseHeaderModifier, &outboundActions, responseHeaders)
+			if err != nil {
+				return nil, err
+			}
+		case gatewayv1.HTTPRouteFilterURLRewrite:
+			return nil, errors.NewErrorNotFound(fmt.Sprintf("Unsupported filter HTTPRouteFilterType %v found", filter.Type))
+		case gatewayv1.HTTPRouteFilterRequestMirror:
+			return nil, errors.NewErrorNotFound(fmt.Sprintf("Unsupported filter HTTPRouteFilterType %v found", filter.Type))
+		case gatewayv1.HTTPRouteFilterExtensionRef:
+			return nil, errors.NewErrorNotFound(fmt.Sprintf("Unsupported filter HTTPRouteFilterType %v found", filter.Type))
+		default:
+			return nil, errors.NewErrorNotFound(fmt.Sprintf("Unknown filter HTTPRouteFilterType %v found", filter.Type))
+		}
+	}
+
+	var policy *ingressv1alpha1.EndpointPolicy
+	enabled := true
+
+	if len(expressions) > 1 {
+		expressions = []string{strings.Join(expressions[:], " || ")}
+	}
+
+	if inboundActions != nil {
+		policy = &ingressv1alpha1.EndpointPolicy{
+			Enabled: &enabled,
+			// NOTE: Mapping each HTTPRouteRule to one Inbound endpoint rule
+			Inbound: []ingressv1alpha1.EndpointRule{
+				{
+					Expressions: expressions,
+					Actions:     *inboundActions,
+					Name:        "Inbound HTTPRouteRule",
+				},
+			},
+		}
+	}
+	if outboundActions != nil {
+		if policy == nil {
+			policy = &ingressv1alpha1.EndpointPolicy{
+				Enabled: &enabled,
+			}
+		}
+
+		policy.Outbound = []ingressv1alpha1.EndpointRule{
+			{
+				Expressions: expressions,
+				Actions:     *outboundActions,
+				Name:        "Outbound HTTPRouteRule",
+			},
+		}
+	}
+
+	return policy, nil
+}
+
+type RemoveHeadersConfig struct {
+	Headers []string `json:"headers"`
+}
+
+type AddHeadersConfig struct {
+	Headers map[string]string `json:"headers"`
+}
+
+func (d *Driver) handleHTTPHeaderFilter(filter *gatewayv1.HTTPHeaderFilter, actions **[]ingressv1alpha1.EndpointAction, requestRedirectHeaders map[string]string) error {
+	if *actions == nil {
+		*actions = &[]ingressv1alpha1.EndpointAction{}
+	}
+	if filter != nil {
+		err := d.handleHTTPHeaderFilterRemove(filter.Remove, actions)
+		if err != nil {
+			return err
+		}
+		err = d.handleHTTPHeaderFilterAdd(filter.Add, actions, requestRedirectHeaders)
+		if err != nil {
+			return err
+		}
+		err = d.handleHTTPHeaderFilterSet(filter, actions, requestRedirectHeaders)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (d *Driver) handleHTTPHeaderFilterRemove(headersToRemove []string, actions **[]ingressv1alpha1.EndpointAction) error {
+	if len(headersToRemove) > 0 {
+		removeHeaders, err := json.Marshal(RemoveHeadersConfig{Headers: headersToRemove})
+		if err != nil {
+			d.log.Error(err, "cannot convert headers to json", "headers", headersToRemove)
+			return err
+		} else {
```

Comment:
no need for else here #nit

Reply:
and further down 🤷

Reply:
nice, also fixed here: https://github.com/ngrok/kubernetes-ingress-controller/pull/343/commits/e99dce670699fdfc565a6eee0faf57c077c94b35